### PR TITLE
feat: migrate v3 analytics

### PR DESCRIPTION
**Pull request by charleenfei**

### CODE REVIEW

**Review by offerijns — APPROVED** [review]
Looks good 👍

File: src/actions/analytics.ts [hunk 1]
```diff
@@ -10,16 +10,9 @@ export function AnalyticsActions<ActionsBase extends Constructor<TinlakeParams>>
...
 
-    // REV: probably can be removed
-    getTotalBalance = async (): Promise<BN> => {
-      return (await this.contract('SHELF').balance()).toBN()
-    }
-
-    // TODO: REV: replace with NAVFeed contract call -- placeholder return for now
     getPrincipal = async (loanId: string): Promise<BN> => {
-      return (await this.contract('SHELF').balance()).toBN()
+      return (await this.contract('FEED').ceiling(loanId)).toBN()
```
Comment (philipstanislaus) [inline src/actions/analytics.ts:14]: Is `ceiling` really `principal` now?

File: src/actions/analytics.ts [hunk 2]
```diff
@@ -137,14 +125,8 @@ export function AnalyticsActions<ActionsBase extends Constructor<TinlakeParams>>
...
 
-    // REV: remove
     getMaxSupplyAmountJunior = async (user: string) => {
-      return (await this.contract('JUNIOR_OPERATOR').maxCurrency(user)).toBN()
-    }
-
-    // REV: remove
-    getMaxRedeemAmountJunior = async (user: string) => {
-      return (await this.contract('JUNIOR_OPERATOR').maxToken(user)).toBN()
+      return (await this.contract('TINLAKE_CURRENCY').balanceOf(user)).toBN()
```
Comment (philipstanislaus) [inline src/actions/analytics.ts:129]: Do we really need this? It is a duplicate of `getCurrencyBalance` in `src/actions/currency.ts`.

File: src/actions/analytics.ts [hunk 3]
```diff
@@ -165,62 +147,26 @@ export function AnalyticsActions<ActionsBase extends Constructor<TinlakeParams>>
...
-          break
-        case 'ALLOWANCE_OPERATOR':
-          maxRedeem = (await this.contract('SENIOR_OPERATOR').maxToken(user)).toBN()
-          break
-        default:
-          maxRedeem = new BN(0)
-      }
-      return maxRedeem
+      if (!this.existsSenior()) return new BN(0)
+      return (await this.contract('TINLAKE_CURRENCY').balanceOf(user)).toBN()
```
Comment (philipstanislaus) [inline src/actions/analytics.ts:152]: Same here.

**Review by philipstanislaus — COMMENTED** [review]
Thanks, added a few comments :)

File: src/actions/analytics.ts (hunk 1 above)
Comment (charleenfei) [reply src/actions/analytics.ts:14]: yes, it is

File: src/actions/analytics.ts (hunk 2 above)
Comment (charleenfei) [reply src/actions/analytics.ts:129]: ah, didn't check currency. i can remove this function

**Pull request merged**